struct_env/struct_env.py

```diff
@@ -5,37 +5,82 @@ class Struct:
 
     def __init__(self, config=None):
         if config is None:
-            config = {"components": 2, "discount_reward": 1, "k_comp": None}
+            config = {"components": 2,
+                      "discount_reward": 1,
+                      "k_comp": None,
+                      "env_type": "uncorrelated"}
+        assert "components" in config and \
+               "discount_reward" in config and \
+               "k_comp" in config and \
+               "env_type" in config, \
+            "Missing env config"
+
         self.ncomp = config["components"]
         self.discount_reward = config["discount_reward"]
-        self.k_comp = self.ncomp-1 if config["k_comp"] is None else config["k_comp"]
+        self.k_comp = self.ncomp - 1 if config["k_comp"] is None \
+            else config["k_comp"]
+        self.env_type = config["env_type"]
         self.time = 0
-        self.ep_length = 30
-        self.nstcomp = 30  # What is this?
-        self.nobs = 2  # What is this?
-        self.actions_per_agent = 3
+        self.ep_length = 30 # Horizon length
+        self.nstcomp = 30  # Crack states (fatigue hotspot damage states)
+        self.nsthyperp = 80 if self.env_type is "correlated" else None  # Gaussian hyperparemeter states 
+        self.nobs = 2  # Total number of observations (crack detected / crack not detected)
+        self.actions_per_agent = 3 
 
-        # obs = 30 per agent + 1 timestep
-        self.obs_per_agent_multi = 31
+        # Uncorrelated obs = 30 per agent + 1 timestep
+        # Correlated obs = 30 per agent + 1 timestep +
+        #                   80 hyperparameter states = 111
+        self.obs_per_agent_multi = 31 if self.env_type is "uncorrelated" else 111
         self.obs_total_single = 30 * self.ncomp + 1
```

Comment:
Should we also add here: self.obs_total_single = 30 * self.ncomp + 81

Reply:
Yes indeed, I'll have a look at the `pymarl_sa_struct` before merging then!